components/hts221/hts221.c

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright (c) 2021 saasaa <mail@saasaa.xyz>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#include "hts221.h"
+
+#include <esp_err.h>
+#include <esp_log.h>
+#include <freertos/FreeRTOS.h>
+#include <i2cdev.h>
+
+#define I2C_FREQ_HZ 400000  // 400kHz
+
+#define CHECK(x)                       \
+  do {                                 \
+    esp_err_t __;                      \
+    if ((__ = x) != ESP_OK) return __; \
+  } while (0)
+
+#define CHECK_ARG(VAL)                      \
+  do {                                      \
+    if (!(VAL)) return ESP_ERR_INVALID_ARG; \
+  } while (0)
+
+static const char *TAG = "hts221";
+
+static calibration_param_t calibration_parameters = {0};
+
+esp_err_t hts221_init_desc(i2c_dev_t *dev, uint8_t addr, i2c_port_t port,
```

Comment:
Please format your source code the way other library sources do. In particular, the opening curly brace must appear on the next line without indentation.

Reply:
Also identation size = 4